### PR TITLE
Add Redis 2.0 package

### PR DESCRIPTION
**Pull request by BrennanConroy**
Sharing code between Redis and StachExchangeRedis packages. Tested the single functional redis test locally.

https://github.com/aspnet/Home/issues/3536

### CODE REVIEW

File: src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackageTags>aspnetcore;dataprotection;redis</PackageTags>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.AspNetCore.DataProtection\Microsoft.AspNetCore.DataProtection.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="StackExchange.Redis.StrongName" Version="$(StackExchangeRedisStrongNamePackageVersion)" />
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.csproj]: Missed a spot

File: test/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.Test/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.Test.csproj [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  <ItemGroup>
+    <Content Include="testconfig.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.DataProtection.Abstractions\Microsoft.AspNetCore.DataProtection.Abstractions.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.DataProtection.Redis\Microsoft.AspNetCore.DataProtection.Redis.csproj" />
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.Test/Microsoft.AspNetCore.DataProtection.StackExchangeRedis.Test.csproj]: Wrong reference

**Comment by Tratcher** [comment]
Also copy over the baseline.netcore.json

**Comment by BrennanConroy** [comment]
Copy or generate a new one? The new one will update the AssemblyIdentity

**Comment by Tratcher** [comment]
Copy I think. Run it to make sure it works. We don't want any regressions from 2.1 here even though this is officially a new package.

**Comment by blowdart** [comment]
I am confused about why StackExchangeRedis?

**Comment by Eilon** [comment]
> I am confused about why StackExchangeRedis?

@blowdart because we had a mail thread about it 😄 For the actual Redis package there's a naming hierarchy: Org.Product.Feature (or whatever). But in our case there's no hierarchy - it's just the OrgProductFeature variation of a package we have.

**Comment by BrennanConroy** [comment]
@anurse @Tratcher 
The public APIs that include redis are `class RedisXmlRepository`, `static class RedisDataProtectionBuilderExtensions` with `PersistKeysToRedis`

Should `PersistKeysToRedis` be changed to `PersistKeysToStackExchangeRedis`? Should any others be changed?

**Comment by natemcmaster** [comment]
`PersistKeysToStackExchangeRedis` seems like an odd API name to me. Sure, StackExchange is the org that implemented the client, but "Redis" is the name of the storage mechanism. We wouldn't name the API `PersistKeysToMicrosoftRedis` if we had implemented the Redis client.

**Comment by Eilon** [comment]
I think `PersistKeysToStackExchangeRedis` is correct (or at least something like it), though it is a mouthful. That it's using StackExchange.Redis is not just an implementation detail - it's the purpose of the API.

And if it was the Microsoft one? I think it probably *should* say MicrosoftRedis, though in practice we often don't do that. E.g. we just say SqlServer and not MicrosoftSqlServer. We're lazy.

**Comment by natemcmaster** [comment]
Ok, I see the thinking. 

We'll need to update docs, by the way. https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/implementation/key-storage-providers. This API was already documented as `PersistKeysToRedis`.

File: src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis/RedisXmlRepository.cs [hunk 3]
```diff
@@ -8,12 +8,12 @@
...
 using Microsoft.AspNetCore.DataProtection.Repositories;
 
-namespace Microsoft.AspNetCore.DataProtection
+namespace Microsoft.AspNetCore.DataProtection.StackExchangeRedis
 {
     /// <summary>
     /// An XML repository backed by a Redis list entry.
     /// </summary>
-    public class RedisXmlRepository: IXmlRepository
+    public class RedisXmlRepository : IXmlRepository
```
[the remark this answers is not in the document]
Comment (Eilon) [reply src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis/RedisXmlRepository.cs]: I could go either way. It's already in a StackExchangeRedis namespace.

[the remark this answers is not in the document]
Comment (natemcmaster) [reply src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis/RedisXmlRepository.cs]: Also don't care, but it seems consistent with all the other renames being done.

[the remark this answers is not in the document]
Comment (muratg) [reply src/Microsoft.AspNetCore.DataProtection.StackExchangeRedis/RedisXmlRepository.cs]: So is this good? I'd like @BrennanConroy can get this in ASAP.

**Comment by BrennanConroy** [comment]
I'll do that tomorrow!

**Pull request merged**